# __Profitable app profiles for the app store and          google play markets__
* ### introduction:
* This project is about making data analysis on various apps on the app store and google play market to check on apps for ratings, downloads, genre, app size, etc.
* our aim in this project is to find the type of free apps available on the app store and play store that attract users the most.


In [1]:

from csv import reader

opened_file = open('googleplaystore.csv')
read_file = reader(opened_file)
android = list(read_file)
android_header = android[0]
android = android[1:]

opened_file = open('AppleStore.csv')
read_file = reader(opened_file)
ios = list(read_file)
ios_header = ios[0]
ios = ios[1:]

## creating a function for printing a data set and the number of rows in it
* in the below cell we will be creating a function which will take 4 inputs.
* they are dataset,start row, end row and a boolean type variable which is initiated as false.
* The function will slice the data set into number of rows we mention in the function. The start row and the end row determine the number of rows we will be working on
* It prints the number of rows we want and also gives information about the number of rows and coloumns in the data set if we wish to.

In [2]:
def explore_data(dataset, start, end, rows_and_columns=False):
    dataset_slice = dataset[start:end]    
    for row in dataset_slice:
        print(row)
        print('\n')
        
    if rows_and_columns:
        print('Number of rows:', len(dataset))
        print('Number of columns:', len(dataset[0]))

* we have now created the function named explore data and now lets test it on the data set we are given
* we will first print the first three rows of android, find the number of rows and coloumns and then use the same function for ios data set

In [3]:
print(android_header)
explore_data(android,0,3,True)

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']
['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['Coloring book moana', 'ART_AND_DESIGN', '3.9', '967', '14M', '500,000+', 'Free', '0', 'Everyone', 'Art & Design;Pretend Play', 'January 15, 2018', '2.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


Number of rows: 10841
Number of columns: 13


In [4]:
print(ios_header)
explore_data(ios,0,3,True)

['id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']
['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 'Games', '38', '5', '18', '1']


Number of rows: 7197
Number of columns: 16


### deleting apps with errors:
* if we observe the 10472 row of android , we can see that the category coloumn of that row is missing and all the coloumns after that are shifted by one place to the left. so there is change in the order of information about the app named 'Life Made WI-Fi Touchscreen Photo Frame'.
* we delete the app because this causes an error when we perform operations with the google play data set
* obseve that after deleting the row the lenght of the android list reduces by one i.e it changes from 10841 to 10840. also the row changes.make sure that you dont run the del code line more than once because it might lead to deleting more than one row.

In [5]:
for i in android_header:
    print(i,end=', ')
print(android[10])
print(len(android))
print(android[10472])
del android[10472]
print(len(android))
print(android[10472])

App, Category, Rating, Reviews, Size, Installs, Type, Price, Content Rating, Genres, Last Updated, Current Ver, Android Ver, ['Text on Photo - Fonteee', 'ART_AND_DESIGN', '4.4', '13880', '28M', '1,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'October 27, 2017', '1.0.4', '4.1 and up']
10841
['Life Made WI-Fi Touchscreen Photo Frame', '1.9', '19', '3.0M', '1,000+', 'Free', '0', 'Everyone', '', 'February 11, 2018', '1.0.19', '4.0 and up']
10840
['osmino Wi-Fi: free WiFi', 'TOOLS', '4.2', '134203', '4.1M', '10,000,000+', 'Free', '0', 'Everyone', 'Tools', 'August 7, 2018', '6.06.14', '4.4 and up']


In [6]:
for j in ios_header:
    print(j,end=', ')

id, track_name, size_bytes, currency, price, rating_count_tot, rating_count_ver, user_rating, user_rating_ver, ver, cont_rating, prime_genre, sup_devices.num, ipadSc_urls.num, lang.num, vpp_lic, 

* if you are having trouble understanding what a column describes visit [documentation](https://www.kaggle.com/ramamet4/app-store-apple-data-set-10k-apps/home)

### Collecting duplicates into a list :
* Here we will be creating 2 lists named unique and duplicate
* we looped through the android data set and segregated it into unique apps and duplicate apps 
* In unique apps we have each app occuring just once and in duplicate apps we have all the apps which occur more than once while we loop thorough the data set
* we wont be removing duplicates randomly. we will keep the apps which have most reviews and remove rest of the duplicates. this is done because the app with most number of reviews is the most recent and updated data of the app

In [7]:
duplicate=[]
unique=[]
for i in android:
    name=i[0]
    if name in unique:
        duplicate.append(name)
    else:
        unique.append(name)
print(len(duplicate))
print(duplicate[0:5])

1181
['Quick PDF Scanner + OCR FREE', 'Box', 'Google My Business', 'ZOOM Cloud Meetings', 'join.me - Simple Meetings']


In [8]:
reviews_max = {}

for app in android:
    name = app[0]
    n_reviews = float(app[3])
    
    if name in reviews_max and reviews_max[name] < n_reviews:
        reviews_max[name] = n_reviews
        
    elif name not in reviews_max:
        reviews_max[name] = n_reviews
print(len(reviews_max))

9659


## cleaning data set
* in the below code cell we cleaned the google play data set which is stored by the name android 
* we first two lists named android_clean and alredy_added.
* the android_clean consists of all the apps with maximum reviews and with no duplicates
* the already_clean list keeps a record of the apps that we already added
* we looped through android list and saved the name of the apps in a variable called name and reviews in a variable called n_reviews.
* we then checked if n_reviews is the same as the number of maximum reviews of the app name (the number can be found in the reviews_max dictionary) and name is not already in the list already_added. if the statement satisfies then:
1. Append the entire row to the android_clean list (which will eventually be a list of list and store our cleaned data set).
2. Append the name of the app name to the already_added list — this helps us to keep track of apps that we already added.
* and finally we checked the length of the android_clean list

In [9]:
android_clean=[]
already_added=[]
for app in android:
    name=app[0]
    n_reviews=float(app[3])
    if n_reviews==reviews_max[name] and name not in already_added:
        android_clean.append(app)
        already_added.append(name)
print(len(android_clean))

9659


## function for english apps
* The code cell below we created a function called eng which takes in a string and returns false if number of non english terms in the name of the app is more than three and true if less than three.

In [10]:
def eng(string):
    c=0
    for i in string:
        if ord(i)>127:
            c+=1
    if c>3:
        return False
    else:
        return True

* we will now collect all the english apps in a list called android_eng and ios_eng respectively

In [11]:
android_eng=[]
ios_eng=[]
for j in android_clean:
    name=j[0]
    if eng(name):
        android_eng.append(j)
for k in ios:
    name=k[1]
    if eng(name):
        ios_eng.append(k)

In [12]:
explore_data(android_eng,0,3,True)
explore_data(ios_eng,0,3,True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


Number of rows: 9614
Number of columns: 13
['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 'G

## collecting all english and free apps:
* we will now store all the free english apps into two lists android_final for google play data set and ios_final for ios app store.

In [13]:
android_final=[]
ios_final=[]
for i in android_eng:
    price=i[7]
    if price=='0':
        android_final.append(i)
for j in ios_eng:
    price=j[4]
    if price=='0.0':
        ios_final.append(j)
explore_data(android_final,0,3,True)
explore_data(ios_final,0,3,True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


Number of rows: 8864
Number of columns: 13
['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 'G

## frequency table

In [14]:
def freq_table(dataset,index):
    table={}
    total=0
    for i in dataset:
        total+=1
        value=i[index]
        if value in table:
            table[value]+=1
        else:
            table[value]=1
    table_p={}
    for key in table:
        percentage=(table[key]/total)*100
        table_p[key]=percentage
    return table_p
def disp_table(dataset,index):
    table=freq_table(dataset,index)
    table_display=[]
    for key in table:
        key_val_tuple=(table[key],key)
        table_display.append(key_val_tuple)
    table_sorted=sorted(table_display,reverse=True)
    for i in table_sorted:
        print(i[1],' : ',i[0])
        

* using the disp_table function for the prime_genre coloumn of the app store.

In [15]:
disp_table(ios_final,-5)

Games  :  58.16263190564867
Entertainment  :  7.883302296710118
Photo & Video  :  4.9658597144630665
Education  :  3.662321539416512
Social Networking  :  3.2898820608317814
Shopping  :  2.60707635009311
Utilities  :  2.5139664804469275
Sports  :  2.1415270018621975
Music  :  2.0484171322160147
Health & Fitness  :  2.0173805090006205
Productivity  :  1.7380509000620732
Lifestyle  :  1.5828677839851024
News  :  1.3345747982619491
Travel  :  1.2414649286157666
Finance  :  1.1173184357541899
Weather  :  0.8690254500310366
Food & Drink  :  0.8069522036002483
Reference  :  0.5586592178770949
Business  :  0.5276225946617008
Book  :  0.4345127250155183
Navigation  :  0.186219739292365
Medical  :  0.186219739292365
Catalogs  :  0.12414649286157665


* we have used the  function for the ios dataset for the prime_genre coloumn and we found that the free english apps consist mainly of gaming applications(58.16 %) .The gaming apps consist of more than half of free english apps.
* following games we have entertainment genre which consists of 7.88 % of the free english apps for the app store data set
* and then we have photo & Video(4.96%), Education(3.66%) and social networking(3.28%)
* however we observe that entertainment apps(Games,Entertainment,Photo & Video,Social Networking,Music,Sports, etc) are more developed by the app developers than other apps for education,shopping, utilities, productivity, lifestyle, etc.
* this analysis is not for the whole app store dataset but for the free english apps.
* however we should keep in mind that fun apps built more by developers that doesnt mean they are most used ones.

In [16]:
disp_table(android_final,1) # category

FAMILY  :  18.907942238267147
GAME  :  9.724729241877256
TOOLS  :  8.461191335740072
BUSINESS  :  4.591606498194946
LIFESTYLE  :  3.9034296028880866
PRODUCTIVITY  :  3.892148014440433
FINANCE  :  3.7003610108303246
MEDICAL  :  3.531137184115524
SPORTS  :  3.395758122743682
PERSONALIZATION  :  3.3167870036101084
COMMUNICATION  :  3.2378158844765346
HEALTH_AND_FITNESS  :  3.0798736462093865
PHOTOGRAPHY  :  2.944494584837545
NEWS_AND_MAGAZINES  :  2.7978339350180503
SOCIAL  :  2.6624548736462095
TRAVEL_AND_LOCAL  :  2.33528880866426
SHOPPING  :  2.2450361010830324
BOOKS_AND_REFERENCE  :  2.1435018050541514
DATING  :  1.861462093862816
VIDEO_PLAYERS  :  1.7937725631768955
MAPS_AND_NAVIGATION  :  1.3989169675090252
FOOD_AND_DRINK  :  1.2409747292418771
EDUCATION  :  1.1620036101083033
ENTERTAINMENT  :  0.9589350180505415
LIBRARIES_AND_DEMO  :  0.9363718411552346
AUTO_AND_VEHICLES  :  0.9250902527075812
HOUSE_AND_HOME  :  0.8235559566787004
WEATHER  :  0.8009927797833934
EVENTS  :  0.7107400

* if we take a look at the googple play dataset we observe that the frequency table of category is quite different from that of app store.
* There are more apps for practical use(family, tools, business, lifestyle, productivity, etc.) compared to app store.
* even in google play dataset we find that fun apps are more than the practical purpose apps.Family category takes up 18.907% but it is less compared to ios app store.
* hence we can say that practical apps are less than the fun aps in both app sotre and google play but the practical apps are more represented in gooogle play than in apps store .

In [17]:
disp_table(android_final,-4) #genre

Tools  :  8.449909747292418
Entertainment  :  6.069494584837545
Education  :  5.347472924187725
Business  :  4.591606498194946
Productivity  :  3.892148014440433
Lifestyle  :  3.892148014440433
Finance  :  3.7003610108303246
Medical  :  3.531137184115524
Sports  :  3.463447653429603
Personalization  :  3.3167870036101084
Communication  :  3.2378158844765346
Action  :  3.1024368231046933
Health & Fitness  :  3.0798736462093865
Photography  :  2.944494584837545
News & Magazines  :  2.7978339350180503
Social  :  2.6624548736462095
Travel & Local  :  2.3240072202166067
Shopping  :  2.2450361010830324
Books & Reference  :  2.1435018050541514
Simulation  :  2.0419675090252705
Dating  :  1.861462093862816
Arcade  :  1.8501805054151623
Video Players & Editors  :  1.7712093862815883
Casual  :  1.7599277978339352
Maps & Navigation  :  1.3989169675090252
Food & Drink  :  1.2409747292418771
Puzzle  :  1.128158844765343
Racing  :  0.9927797833935018
Role Playing  :  0.9363718411552346
Libraries & D

* The difference between the Genres and the Category columns is not crystal clear, but one thing we can notice is that the Genres column is much more granular (it has more categories). We're only looking for the bigger picture at the moment, so we'll only work with the Category column moving forward.

* Up to this point, we found that the App Store is dominated by apps designed for fun, while Google Play shows a more balanced landscape of both practical and for-fun apps. Now we'd like to get an idea about the kind of apps that have most users.

# most popular apps on app store
* To find the most popular app on the app store we will need to find the apps 
 with most installs. we have a coloumn in google play data set but there is  
 no such coloumn in app store. as an alternative we will take number of user rating for finding popular apps in ios data set
* first we store the frequency table of ios data set in genres_ios variable for the genres coloumn.then we will loop through frequency table and if genre_app is equal to genre we will save the rating into a variable named no_rating and also incremment the length value by one.to find the average rating we need to total rating by length of genre. and then finally print the genre and its coressponding average rating.

In [18]:
genres_ios=freq_table(ios_final,-5)
for genre in  genres_ios:
    total_r=0
    len_genre=0
    for i in ios_final:
        genre_app=i[-5]
        if genre_app==genre:
            no_rating=float(i[5])
            total_r+=no_rating
            len_genre+=1
    avg_rating_ios=(total_r/len_genre)
    print(genre, ':' ,avg_rating_ios)

News : 21248.023255813954
Catalogs : 4004.0
Reference : 74942.11111111111
Social Networking : 71548.34905660378
Lifestyle : 16485.764705882353
Weather : 52279.892857142855
Utilities : 18684.456790123455
Medical : 612.0
Finance : 31467.944444444445
Travel : 28243.8
Shopping : 26919.690476190477
Food & Drink : 33333.92307692308
Productivity : 21028.410714285714
Book : 39758.5
Photo & Video : 28441.54375
Navigation : 86090.33333333333
Entertainment : 14029.830708661417
Sports : 23008.898550724636
Education : 7003.983050847458
Games : 22788.6696905016
Health & Fitness : 23298.015384615384
Music : 57326.530303030304
Business : 7491.117647058823


* looking at the above average rating we can see that social networking has a huge average rating. but if we look at user ratings of each app we can see that facebook, instagram, whatsapp messenger take up more than 70% of the user ratings for that genre. same with music apps,reference apps,navigaton apps.

In [19]:
for i in ios_final:
    if i[-5]=='Social Networking':
        print(i[1]," : ",i[5])

Facebook  :  2974676
Pinterest  :  1061624
Skype for iPhone  :  373519
Messenger  :  351466
Tumblr  :  334293
WhatsApp Messenger  :  287589
Kik  :  260965
ooVoo – Free Video Call, Text and Voice  :  177501
TextNow - Unlimited Text + Calls  :  164963
Viber Messenger – Text & Call  :  164249
Followers - Social Analytics For Instagram  :  112778
MeetMe - Chat and Meet New People  :  97072
We Heart It - Fashion, wallpapers, quotes, tattoos  :  90414
InsTrack for Instagram - Analytics Plus More  :  85535
Tango - Free Video Call, Voice and Chat  :  75412
LinkedIn  :  71856
Match™ - #1 Dating App.  :  60659
Skype for iPad  :  60163
POF - Best Dating App for Conversations  :  52642
Timehop  :  49510
Find My Family, Friends & iPhone - Life360 Locator  :  43877
Whisper - Share, Express, Meet  :  39819
Hangouts  :  36404
LINE PLAY - Your Avatar World  :  34677
WeChat  :  34584
Badoo - Meet New People, Chat, Socialize.  :  34428
Followers + for Instagram - Follower Analytics  :  28633
GroupMe  :  

* so i would recommened an app for education purpose which will focus on teaching to a particular age group of students which is totally free and gains money only by in app ads.This app should contain teaching via video with using grapics so that the student can understand the concept rather then by hearting the concepts

# most popular apps in android :
* Take a look at the android dataset we observe that we have a column for installs in the dataset. 
* this makes our work easier to find out the popular apps.
* we will first use the frequency table for the category column of the android dataset and store it in a variable called genres_an
* the we will loop through genres_an and find if genre_app is equal to genre
* if so we will store the value of installs for that app in total_in variable and increase the length of the genre by one
* find avreage installs for each genre and save it in avg_ins_android variable.
* print the genre and its corresponding avg installs

In [20]:
genres_an=freq_table(android_final,1)
for genre in genres_an:
    total_in=0
    len_genre=0
    for i in android_final:
        genre_app=i[1]
        if genre_app==genre:
            n_in=i[5]
            n_in=n_in.replace('+','')
            n_in=n_in.replace(',','')
            total_in+=float(n_in)
            len_genre+=1
    avg_ins_android=(total_in/len_genre)
    print(genre ,' : ',avg_ins_android)

HEALTH_AND_FITNESS  :  4188821.9853479853
ENTERTAINMENT  :  11640705.88235294
LIFESTYLE  :  1437816.2687861272
TRAVEL_AND_LOCAL  :  13984077.710144928
HOUSE_AND_HOME  :  1331540.5616438356
PRODUCTIVITY  :  16787331.344927534
SHOPPING  :  7036877.311557789
AUTO_AND_VEHICLES  :  647317.8170731707
TOOLS  :  10801391.298666667
BOOKS_AND_REFERENCE  :  8767811.894736841
COMICS  :  817657.2727272727
PHOTOGRAPHY  :  17840110.40229885
PARENTING  :  542603.6206896552
DATING  :  854028.8303030303
FINANCE  :  1387692.475609756
EDUCATION  :  1833495.145631068
FOOD_AND_DRINK  :  1924897.7363636363
GAME  :  15588015.603248259
BUSINESS  :  1712290.1474201474
BEAUTY  :  513151.88679245283
FAMILY  :  3695641.8198090694
NEWS_AND_MAGAZINES  :  9549178.467741935
EVENTS  :  253542.22222222222
VIDEO_PLAYERS  :  24727872.452830188
MAPS_AND_NAVIGATION  :  4056941.7741935486
LIBRARIES_AND_DEMO  :  638503.734939759
SOCIAL  :  23253652.127118643
PERSONALIZATION  :  5201482.6122448975
MEDICAL  :  120550.6198083067

* if you take a look at each of the genres we can see that social category has one of the most intalls of all

In [21]:
for i in android_final:
    if i[1]=='SOCIAL':
        print(i[0],' : ',i[5])

Facebook  :  1,000,000,000+
Facebook Lite  :  500,000,000+
Tumblr  :  100,000,000+
Social network all in one 2018  :  100,000+
Pinterest  :  100,000,000+
TextNow - free text + calls  :  10,000,000+
Google+  :  1,000,000,000+
The Messenger App  :  1,000,000+
Messenger Pro  :  1,000,000+
Free Messages, Video, Chat,Text for Messenger Plus  :  1,000,000+
Telegram X  :  5,000,000+
The Video Messenger App  :  100,000+
Jodel - The Hyperlocal App  :  1,000,000+
Hide Something - Photo, Video  :  5,000,000+
Love Sticker  :  1,000,000+
Web Browser & Fast Explorer  :  5,000,000+
LiveMe - Video chat, new friends, and make money  :  10,000,000+
VidStatus app - Status Videos & Status Downloader  :  5,000,000+
Love Images  :  1,000,000+
Web Browser ( Fast & Secure Web Explorer)  :  500,000+
SPARK - Live random video chat & meet new people  :  5,000,000+
Golden telegram  :  50,000+
Facebook Local  :  1,000,000+
Meet – Talk to Strangers Using Random Video Chat  :  5,000,000+
MobilePatrol Public Safety A

* but when we dig further into the genre we will find that most of the installs on this category are of facebook,instagram,messenger,tiktok,google+, etc. these 5 to 10 apps take up most of the installs in this category
* same with communication,photography,tools, etc.

In [22]:
under_100_m=[]
for i in android_final:
    if i[1]=='SOCIAL':
        n_in=i[5]
        n_in=n_in.replace(',','')
        n_in=n_in.replace('+','')
        n_in=float(n_in)
        if n_in<100000000:
            under_100_m.append(n_in)
print(sum(under_100_m)/len(under_100_m))

3084582.5201793723


* when we remove the 100 millon installs apps the average install of the category falls drastically
* by this we understand that the only few giants like facebook,google+,instagram,tiktok,etc. take up most of the installs in google play

In [23]:
for i in android_final:
    if i[1]=='EDUCATION':
        print(i[0],' : ',i[5])
        

English Communication - Learn English for Chinese (Learn English for Chinese)  :  100,000+
Khan Academy  :  5,000,000+
Ai La Trieu Phu - ALTP Free  :  100,000+
Learn Spanish - Español  :  1,000,000+
Speed Reading  :  500,000+
English for beginners  :  1,000,000+
Mermaids  :  5,000,000+
Learn Japanese, Korean, Chinese Offline & Free  :  1,000,000+
Kids Mode  :  500,000+
Dinosaurs Coloring Pages  :  500,000+
Cars Coloring Pages  :  1,000,000+
Math Tricks  :  10,000,000+
Learn English Words Free  :  5,000,000+
Japanese / English one-shop search dictionary - Free Japanese - English - Japanese dictionary application  :  50,000+
English speaking texts  :  1,000,000+
Thai Handwriting  :  1,000,000+
THAI DICT 2018  :  1,000,000+
Kanji test · Han search Kanji training (free version)  :  1,000,000+
Flippy Campus - Buy & sell on campus at a discount  :  500,000+
Free intellectual training game application |  :  1,000,000+
ABC Preschool Free  :  5,000,000+
PINKFONG Baby Shark  :  1,000,000+
Englis

* education category has many apps in it but  It seems there's still a small number of extremely popular apps that skew the average:         

In [24]:
for i in android_final:
    if i[1]=='EDUCATION' and (i[5] =='10,000,000+'):
        print(i[0], ':', i[5])

Math Tricks : 10,000,000+
Lumosity: #1 Brain Games & Cognitive Training App : 10,000,000+
Quizlet: Learn Languages & Vocab with Flashcards : 10,000,000+
Learn English with Wlingua : 10,000,000+
Remind: School Communication : 10,000,000+
Google Classroom : 10,000,000+
ClassDojo : 10,000,000+
Learn languages, grammar & vocabulary with Memrise : 10,000,000+


* we can clearly see that maximum downloads is just 10 millon and that too we just have 8 apps in that range.
* so clearly there is no app which is popular in this category
* I would recommend to come with an app which is effectively teaches the students through videos, online books and use graphics to make the students understand the concepts in a easier and effective way.

In [25]:
for i in android_final:
    if i[1]=='EDUCATION' and (i[5] =='1,000,000+'
                             or i[5]=='500,000+'
                             or i[5]=='100,000+'):
        print(i[0], ':', i[5])

English Communication - Learn English for Chinese (Learn English for Chinese) : 100,000+
Ai La Trieu Phu - ALTP Free : 100,000+
Learn Spanish - Español : 1,000,000+
Speed Reading : 500,000+
English for beginners : 1,000,000+
Learn Japanese, Korean, Chinese Offline & Free : 1,000,000+
Kids Mode : 500,000+
Dinosaurs Coloring Pages : 500,000+
Cars Coloring Pages : 1,000,000+
English speaking texts : 1,000,000+
Thai Handwriting : 1,000,000+
THAI DICT 2018 : 1,000,000+
Kanji test · Han search Kanji training (free version) : 1,000,000+
Flippy Campus - Buy & sell on campus at a discount : 500,000+
Free intellectual training game application | : 1,000,000+
PINKFONG Baby Shark : 1,000,000+
English words application mikan : 500,000+
Learn English for beginners : 500,000+
Listen and learn English in seven days : 100,000+
Learn English from Persian: Persian to English : 100,000+
Udemy - Online Courses : 1,000,000+
edX - Online Courses by Harvard, MIT & more : 1,000,000+
Memorado - Brain Games : 1,

* when we look at apps with 1 millon or less installs we find numerous apps and this shows that this category is develloping but because most of the apps in education are paid and still not up to the mark we find less numbre of installs for these.
* so coming up with an educational app which is free of cost and approches teaching in an interesting and effective way will be a great idea and will be a profitable app for both app store and gooogple play and also work towards a good cause of helping students in their academics

# Conclusions
* In this project, we analyzed data about the App Store and Google Play mobile apps with the goal of recommending an app profile that can be profitable for both Google play and App store market.
* **we concluded that an educational app could be profitable for google play and app store markets.The markets have a lot of educational apps but only a few of them are a bit popular. these app are not that succesful because they cost you money to teach after you download them. so coming up with an educational app which id free of cost and approches teaching in an intersting and effective way we will a great idea for both markets and also work towards a good cause of helping students in their academics** 